src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1282,6 +1282,19 @@ int LinearScan::BuildNode(GenTree* tree)
             srcCount = BuildSelect(tree->AsOp());
             break;
 
+#ifdef SWIFT_SUPPORT
+        case GT_SWIFT_ERROR:
+            srcCount = 0;
+            assert(dstCount == 1);
+
+            // After a Swift call potentially trashes the error register,
+            // the register can be used again only if there is a GT_SWIFT_ERROR node to consume it
+            // (i.e. the register's value is saved to a SwiftError)
+            addRefsForPhysRegMask(RBM_SWIFT_ERROR, currentLoc, RefTypeKill, true);
+            BuildDef(tree, RBM_SWIFT_ERROR);
```

Comment:
`RBM_SWIFT_ERROR` as the candidate set here is strictly speaking not necessary, but it might produce better code.

Reply:
Could you elaborate on how we could do this differently, and why this might produce better code, please? I got this from your async2 prototype.

Reply:
The candidate set here is saying that the `GT_SWIFT_ERROR` node itself always needs `r12` as its destination register. But in reality any assignment picked by LSRA is fine for the `GT_SWIFT_ERROR` node's value -- it just means that we have to insert a copy from `r12` into the register that LSRA picked when we generate the `GT_SWIFT_ERROR node (that's exactly what `genCodeForPhysReg` is doing).
The codegen might be better because requiring LSRA to pick `r12` means we can always skip that copy during codegen. Normally LSRA utilizes various preferencing heuristics to try to make this the case, even without specifying a "hard" requirement on the destination candidates, but nothing here is communicating this preference to LSRA. I think you could do that instead by:
```cpp
BuildDef(tree)->getInterval()->mergeRegisterPreferences(RBM_SWIFT_ERROR);
```
It would be slightly more faithful to the actual constraints we're dealing with here, but I don't think it necessitates a change (in particular since we ensure that `r12` is busy until the point of the def we know it's going to be available to assign).

Reply:
I see, thanks for the explanation! I'll update the comment to clarify this decision.